src/vs/workbench/parts/extensions/electron-browser/extensionsWorkbenchService.ts

```diff
@@ -108,7 +108,7 @@ class Extension implements IExtension {
 			return this.local.changelogUrl;
 		}
 
-		return ''; // Hopefully we will change this once the gallery will support that.
+		return this.gallery && this.gallery.assets.changelog;
```

Comment:
This was the wrong place. You should do it in the `hasChangelog` and `getChangelog`. While you're at it, just delete the `readmeUrl` and `changelogUrl`; the former was an old implementation which I forgot to delete.